### PR TITLE
fix: make createUrl more robust

### PR DESCRIPTION
**Pull request by stepan662**
Closing #3458 

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved URL creation to handle invalid base values without causing errors, ensuring more reliable URL generation in edge cases.
- **Chores**
  - Upgraded Cypress testing framework to version 14.4.1 and updated test scripts to consistently use Edge as the test browser.
  - Removed Cypress browser extension plugin and updated test configurations accordingly.
  - Skipped certain UI screenshot tests due to known browser compatibility issues.

### CODE REVIEW

File: e2e/scripts/startCypress.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/sh
 set -x
 
-ARGS="--browser chrome --headed"
+ARGS="--browser edge --headed"
```
Comment (JanCizmar) [inline e2e/scripts/startCypress.sh:4]: Edge?

**Review by JanCizmar — APPROVED** [review]
